### PR TITLE
QoL changes

### PR DESCRIPTION
**Pull request by MesterMan03**
+ Alerts from ws are now saved.
+ Toast text is nicer.
~ Appeal button doesn't show up if the punishment is appealed.
~ Better TypeScript compatibility.
~ Changed toast text timings for better readability.

### CODE REVIEW

**Review by noClaps — CHANGES_REQUESTED** [review]
I think it would be better to have all of the alerts code in a single `alerts.ts` file, while keeping the `types.ts` only for types. As we expand alerts functionality, we can split the code into more files and put it all into an `alerts/` directory.

File: src/scripts/types.ts [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
+  : never;
+
+export interface Alert<T extends AlertType> {
+  type: T;
+  data: AlertData<T>;
+}
+
+export function isAppealAlert(alert: Alert<AlertType>): alert is Alert<"appeal"> {
+  return alert.type === "appeal";
 }
```
Comment (noClaps) [inline src/scripts/types.ts:37]: I think this function is pointless, since this check can easily be made inline.

File: src/pages/alerts.astro [hunk 2]
```diff
@@ -41,17 +41,23 @@ import "../styles/alerts.css";
...
     localStorage.setItem("alertHistory", JSON.stringify(alertHistory));
 
     const alertElement = document.createElement("div");
-    alertElement.className = alert.data.status;
-    alertElement.innerHTML = alert.data.reason;
+
+    if (isAppealAlert(alert)) {
+      // using className like this will work for now, but with more alert types it'll become spaghetti code
+      alertElement.className = alert.data.status;
+      alertElement.innerHTML = alert.data.reason;
```
Comment (noClaps) [inline src/pages/alerts.astro:57]: You can use:

```ts
alertElement.classList.add(alert.data.status)
```

to achieve this. If you want the different alert types to be styled differently then you have to modify the element itself in some way for the CSS to pick it up and style it, whether that's using HTML datasets, or classes.

**Comment by MesterMan03** [comment]
> I think this function is pointless, since this check can easily be made inline. 

it is pointless *for now*, but when we add more alert types, we'll have to add it anyway

**Comment by noClaps** [comment]
You also haven't added the `appealed` prop to the main `pages/punishments.astro` page.

**Comment by noClaps** [comment]
> > I think this function is pointless, since this check can easily be made inline.
> 
> it is pointless _for now_, but when we add more alert types, we'll have to add it anyway

Yes, but if those are also `return alert.type === [alert type]` then that's still a check that can be made inline.

**Review by noClaps — APPROVED** [review]

**Comment by noClaps** [comment]
LGTM

**Pull request merged**